### PR TITLE
Uniparc uniprotkb style entry

### PR DESCRIPTION
**Pull request by dlrice**
## Purpose
Create a subentry for uniparc xref

## Testing
Snaphots generated

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/uniparc/components/entry/__tests__/__snapshots__/XRefsSection.spec.tsx.snap [hunk 1]
```diff
@@ -89,15 +89,7 @@ exports[`SequenceSection component should render the SequenceSection properly an
...
-                <td>
-                  <span>
-                    <a
-                      href="/uniprotkb/P07612/entry"
-                    >
-                      P07612
-                    </a>
-                  </span>
-                </td>
+                <td />
```
Comment (aurel-l) [inline src/uniparc/components/entry/__tests__/__snapshots__/XRefsSection.spec.tsx.snap:92]: Something might be off here if it's missing content

File: src/uniparc/components/sub-entry/SubEntryOverview.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+type Props = {
+  data: UniParcSubEntryUIModel;
+};
+
+const SubEntryOverview = ({ data }: Props) => {
+  const dataDB = useDataApi<DataDBModel>(
+    apiUrls.configure.allDatabases(Namespace.uniparc)
+  );
+  const [proteomeId, component] =
```
Comment (aurel-l) [inline src/uniparc/components/sub-entry/SubEntryOverview.tsx:52]: Maybe a `let [proteomeId, component] = [];` followed by ifs might be clearer to read

File: src/uniparc/components/sub-entry/SubEntryStructureSection.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    )
+);
+
+type Props = {
+  data: UniParcSubEntryUIModel['subEntry'];
+};
+
+// TODO: this only works for uniprotkb entries but I'm not showing these here
+export const hasStructure = (data: UniParcSubEntryUIModel['subEntry']) =>
+  data.isUniprotkbEntry && data.active && data.id;
```
Comment (aurel-l) [inline src/uniparc/components/sub-entry/SubEntryStructureSection.tsx:22]: Not sure if this logic is valid always.
Some UniProtKB entries don't have structures, and we'll still want to show the structures for the UniProtKB entries that will be removed soon and also for external IDs as eventually it should be based on the sequence. The issue is that we can't query the 3D Beacons API like this yet, so I guess it'll do for now

File: src/uniparc/components/sub-entry/SubEntry.tsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+          }
+          id={TabLocation.FeatureViewer}
+        >
+          {smallScreen ? (
+            <Redirect
+              to={getSubEntryPath(
+                Namespace.uniparc,
+                match.params.accession,
+                TabLocation.Entry
+              )}
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/uniparc/components/sub-entry/SubEntry.tsx]: Fixed

File: src/uniparc/config/UniParcXRefsColumnConfiguration.tsx [hunk 5]
```diff
@@ -99,47 +98,62 @@ UniParcXRefsColumnConfiguration.set(UniParcXRefsColumn.database, {
...
     let cell: ReactNode = xref.id;
     if (
-      xref.database === XRefsInternalDatabasesEnum.REVIEWED ||
-      xref.database === XRefsInternalDatabasesEnum.UNREVIEWED ||
-      xref.database === 'UniProtKB/Swiss-Prot protein isoforms'
+      (xref.database === XRefsInternalDatabasesEnum.REVIEWED ||
+        xref.database === XRefsInternalDatabasesEnum.UNREVIEWED ||
+        xref.database === 'UniProtKB/Swiss-Prot protein isoforms') &&
+      !xref.database.includes('isoforms') &&
+      xref.active
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/uniparc/config/UniParcXRefsColumnConfiguration.tsx]: Reverted for now.

File: src/uniparc/components/entry/__tests__/__snapshots__/XRefsSection.spec.tsx.snap (hunk 1 above)
Comment (dlrice) [reply src/uniparc/components/entry/__tests__/__snapshots__/XRefsSection.spec.tsx.snap]: Content now back.

File: src/uniparc/components/sub-entry/SubEntryOverview.tsx (hunk 2 above)
Comment (dlrice) [reply src/uniparc/components/sub-entry/SubEntryOverview.tsx]: Done.

File: src/uniparc/components/sub-entry/SubEntryStructureSection.tsx (hunk 3 above)
Comment (dlrice) [reply src/uniparc/components/sub-entry/SubEntryStructureSection.tsx:28]: Added this comment to the code.

**Review by aurel-l — APPROVED** [review]

**Pull request merged**